socs/agents/orientalmotor_blh/agent.py

```diff
@@ -0,0 +1,285 @@
+                session.data.update({'timestamp': current_time})
+
+                time.sleep(sleep_time)
+
+            self.agent.feeds['motor'].flush_buffer()
+
+        return True, 'Acquisition exited cleanly.'
+
+    def stop_acq(self, session, params=None):
+        """
+        Stops the data acquisiton.
+        """
+        self.agent.start('stop_rotation')
+        while self.speed > 0.1:
+            time.sleep(1)
+
+        if self.take_data:
+            self.take_data = False
+            return True, 'requested to stop taking data.'
+
+        return False, 'acq is not currently running.'
+
+    def set_values(self, session, params=None):
+        '''A task to set parameters for BLH motor driver
+
+        Parameters
+        ----------
+        speed : int
+            Motor rotation speed in RPM
+        accl_time : float
+            Acceleration time
+        decl_time : float
+            Deceleration time
+        '''
+        if params is None:
+            params = {}
+
+        with self.lock.acquire_timeout(3, job='set_values') as acquired:
+            if not acquired:
+                self.log.warn('Could not start set_values because '
+                              f'{self.lock.job} is already running')
+                return False, 'Could not acquire lock.'
+
+            speed = params.get('speed')
+            if speed is not None:
+                self._blh.set_speed(speed)
+
+            accl_time = params.get('accl_time')
+            if accl_time is not None:
+                self._blh.set_accl_time(accl_time, accl=True)
+
+            decl_time = params.get('decl_time')
+            if decl_time is not None:
+                self._blh.set_accl_time(decl_time, accl=False)
+
+        return True, 'Set values for BLH'
+
+    def start_rotation(self, session, params=None):
+        '''Start rotation
+
+        Parameters
+        ----------
+        forward : bool, default True
+            Move forward if True
+        '''
+        if params is None:
+            params = {}
+
+        if not self.take_data:
+            self.agent.start('acq')
+            for _ in range(ACQ_TIMEOUT):
+                if self.take_data:
+                    break
+                time.sleep(0.1)
+
+        if not self.take_data:
+            return False, 'Could not start acq.'
+
+        with self.lock.acquire_timeout(3, job='set_values') as acquired:
+            if not acquired:
+                self.log.warn('Could not start set_values because '
+                              f'{self.lock.job} is already running')
+                return False, 'Could not acquire lock.'
+
+            if not self.take_data:
+                return False, 'acq is not currently running.'
+
+            forward = params.get('forward')
+            if forward is None:
+                forward = True
+
+            self._blh.start(forward=forward)
+
+        return True, 'BLH rotation started.'
```

Comment:
You should handle the case where the start command failed to run properly, thus the return statement that the rotation started would be incorrect. See my comment in `drivers.py`.